apps/website/src/pages/api/short-links/track-click.ts

```diff
@@ -0,0 +1,30 @@
+import type { NextApiRequest, NextApiResponse } from "next";
+import { prisma } from "@/server/db/client";
+import { env } from "@/env/index.mjs";
+
+//API for short links tracking
+export default async function handler(
+  req: NextApiRequest,
+  res: NextApiResponse,
+) {
+  const body = JSON.parse(req.body);
+  const id = body.id;
+  if (body.secret === env.SHORT_LINKS_TRACKING_SECRET) {
+    const query = await prisma.shortLinksTracking.findFirst({
+      where: { id: id },
+    });
+    if (query) {
+      await prisma.shortLinksTracking.update({
+        where: { id: id },
+        data: {
+          clicks: {
+            increment: 1,
+          },
+        },
+      });
```

Comment:
Right now the code is missing the creation of the tracking entry. I think this was previously done in the trpc procedure addClick, but did not make it to the api. Maybe you can use an upsert (combined insert and update) for this?

```typescript
prisma.shortLinksTracking.upsert({
  where: { id },
  create: { id, clicks: 1 },
  update: { clicks: { increment: 1 } },
})
```